mlflow/pipelines/utils/execution.py

```diff
@@ -270,6 +272,102 @@ def _get_step_output_directory_path(execution_directory_path: str, step_name: st
     )
 
 
+class _ExecutionPlan:
+
+    _MSG_PREFIX = "# Run MLP step: "
+    _FORMAT_STEPS_CACHED = "%s: No changes. Skipping."
+    _FULL_TRAINING_DAG = ["ingest", "split", "transform", "train", "evaluate", "register"]
```

Comment:
Rather than hardcoding the step names, I'd prefer if we reference the `.name` attributes of the ordered pipeline steps passed to `mlflow.pipelines.utils.execution.run_pipeline_step()` here: https://github.com/mlflow/mlflow/blob/78d7504ddf4e93ce768f40b990e28b91dd340ce7/mlflow/pipelines/utils/execution.py#L20.

We can use the `.index()` method of the list to determine the name of the first executed step and then assume that the prior steps are cached.